extensions/standard-processors/processors/SegmentContent.h

```diff
@@ -0,0 +1,78 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string_view>
+#include <utility>
+
+#include "FlowFileRecord.h"
+#include "core/ProcessSession.h"
+#include "core/Processor.h"
+#include "core/PropertyDefinition.h"
+#include "core/PropertyDefinitionBuilder.h"
+#include "core/RelationshipDefinition.h"
+#include "utils/Export.h"
+
+namespace org::apache::nifi::minifi::processors {
+
+class SegmentContent final : public core::Processor {
+ public:
+  explicit SegmentContent(const std::string_view name, const utils::Identifier& uuid = {}) : Processor(name, uuid) {}
+
+  EXTENSIONAPI static constexpr auto Description = "Segments a FlowFile into multiple smaller segments on byte boundaries.";
+
+  EXTENSIONAPI static constexpr auto SegmentSize =
+      core::PropertyDefinitionBuilder<2>::createProperty("Segment Size")
+        .withDescription("The maximum data size in bytes for each segment")
+        .isRequired(true)
+        .supportsExpressionLanguage(true)
+        .build();
```

Comment:
```
.withPropertyType(core::StandardPropertyTypes::DATA_SIZE_TYPE)
```
should be added to allow validation.

Reply:
I've skipped the validation because it doesnt play nice with expression language.
see ExpressionLanguageSupport test

It probably should and I've found some code that sets this property validator and claims to support expression language. e.g FetchAzureDataLakeStorage::NumberOfRetries

I created a [Jira](https://issues.apache.org/jira/browse/MINIFICPP-2478) and we can investigate what to do.